src/Meshes/Meshes.jl

```diff
@@ -132,4 +134,113 @@ struct EquiangularCubedSphereMesh{FT} <: AbstractMesh{FT}
     n::Int64
 end
 
+"""
+    TensorProductMesh(domain::RectangleDomain, n::Integer)
+
+A tensor-product `AbstractMesh` of `domain` with `n1` elements in dimension 1, and `n2`
+in dimension 2.
+"""
+struct TensorProductMesh{FT, RD <: RectangleDomain{FT}} <: AbstractMesh{FT}
```

Comment:
can we move all the TensorProduct-specific stuff to a new file? I'd like the keep `src/Meshes/Meshes.jl` so that it only contains code for generic meshes

Reply:
Yes, but this would just be another generic mesh constructor, like we have already the `EquispacedLineMesh`, `EquispacedRectangleMesh` ones in the same file, and their corresponding `show` methods. Why would this fit in another category?

Reply:
Because they're fairly small. This adds 200 lines, which probably makes sense to be in a different file.

Reply:
okay

Reply:
Done in [`82f89ed`](https://github.com/CliMA/ClimaCore.jl/pull/82/commits/82f89ed04ce33546240e0489473b174aad391484)